### PR TITLE
Code optimization for uprobe.py test script.

### PR DESCRIPTION
**Pull request by SamirMulani**
Code optimization for uprobe.py,
The enable_uprobes and disable_uprobes functions have been removed and a single function has been added to control uprobes.
Added support for handling system exceptions while writing data into the /sys/kernel/debug/tracing/ uprobe_events and other files

Signed-off-by: Samir Mulani <samir@linux.vnet.ibm.com>

### CODE REVIEW

File: ras/uprobe.py [hunk 1]
```diff
@@ -36,27 +35,35 @@ def run_cmd(self, cmd):
...
+            self.uprobes_events_fs = os.path.join(self.debugfs, uprobe_eve)
+            genio.write_one_line(self.tracefs, "")
+            genio.write_one_line(self.uprobes_events_fs, "")
+        except IOError:
+            self.cancel("There is issue with kernel system resources")
 
-    def disable_uprobes(self):
-        genio.write_one_line(self.uprobes_event_fs, "0")
-        genio.write_one_line(self.uprobes_enable_fs, "0")
+    def ena_dis_uprobes(self, prob_value):
```
Comment (Naresh-ibm) [inline ras/uprobe.py:57]: @SamirMulani can you please add docstrings to all the methods.
I don't see to any of the method. having doctrings methods are more self explanatory :-)

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: ras/uprobe.py [hunk 2]
```diff
@@ -70,23 +103,33 @@ def setUp(self):
...
+        for cmd in cmd_list:
+            self.run_cmd(cmd)
+        cmd_trace = "cat /sys/kernel/debug/tracing/trace"
+        if "u_malloc" not in (self.run_cmd_out(cmd_trace)):
             self.fail("Uprobe probe was not hit.")
-        self.disable_uprobes()
+        self.ena_dis_uprobes(self.disable_prob)
         self.clear_trace()
 
     def test(self):
```
Comment (Naresh-ibm) [inline ras/uprobe.py:135]: @SamirMulani please add here too.

File: ras/uprobe.py [hunk 3]
```diff
@@ -70,23 +103,33 @@ def setUp(self):
                 self.cancel('%s is needed for the test to be run' % package)
 
     def execute_test(self):
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:106]: extra line is not require. please check this and remove it.

File: ras/uprobe.py [hunk 4]
```diff
@@ -70,23 +103,33 @@ def setUp(self):
...
+        """
+        -> This function is responsible to insert tracepoints in the kernel.
+        -> It will enable and disable the uprobes.
+        """
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:111]: here too.

File: ras/uprobe.py [hunk 5]
```diff
@@ -36,27 +35,61 @@ def run_cmd(self, cmd):
...
+        try:
+            genio.write_one_line(self.uprobes_enable_fs, prob_value)
+            genio.write_one_line(self.uprobes_event_fs, prob_value)
+        except IOError:
+            self.cancel("There is issue with kernel system resources")
 
     def setUp(self):
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:86]: extra line not required

File: ras/uprobe.py [hunk 6]
```diff
@@ -36,27 +35,61 @@ def run_cmd(self, cmd):
...
+        except IOError:
+            self.cancel("There is issue with kernel system resources")
+
+    def ena_dis_uprobes(self, prob_value):
+        """
+        Enable and Disable the uprobes.
+        ------------
+        Enable_uprobes:
+        prob_value = 1
+        ------------
```
Comment (Naresh-ibm) [inline ras/uprobe.py:74]: I feel these cosmetic line are not required here as your already adding new line for better visualization.

File: ras/uprobe.py [hunk 7]
```diff
@@ -36,27 +35,61 @@ def run_cmd(self, cmd):
...
-        genio.write_one_line(self.uprobes_events_fs, "")
-
-    def enable_uprobes(self):
-        genio.write_one_line(self.uprobes_enable_fs, "1")
-        genio.write_one_line(self.uprobes_event_fs, "1")
-
-    def disable_uprobes(self):
-        genio.write_one_line(self.uprobes_event_fs, "0")
-        genio.write_one_line(self.uprobes_enable_fs, "0")
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:48]: extra line not required

File: ras/uprobe.py [hunk 8]
```diff
@@ -36,27 +35,61 @@ def run_cmd(self, cmd):
 
     @staticmethod
     def run_cmd_out(cmd):
-        return process.system_output(cmd, shell=True, ignore_status=True,
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:38]: same here, extra line not required.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: ras/uprobe.py [hunk 9]
```diff
@@ -68,26 +65,22 @@ def clear_trace(self):
     def ena_dis_uprobes(self, prob_value):
         """
         Enable and Disable the uprobes.
-        ------------
         Enable_uprobes:
         prob_value = 1
-        ------------
+
```
Comment (Naresh-ibm) [inline ras/uprobe.py:70]: you missed this :-)

**Comment by Naresh-ibm** [comment]
@PraveenPenguin its looks good to me.
test case wise if it is ok, then we can merge this.

**Comment by PraveenPenguin** [comment]
@SamirMulani can you please wrap both the commits in single logical commit

**Comment by Naresh-ibm** [comment]
LGTM.
@PraveenPenguin we can merge this if nothing from your end.

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**